server/services/video.service.js

```diff
@@ -0,0 +1,178 @@
+  const video = await getVideoById(updatedVideo.videoId).catch((error) => {
+    throw new ErrorHandler(httpStatus.INTERNAL_SERVER_ERROR, 'Internal Server Error', error);
+  });
+  if (!video) {
+    throw new ErrorHandler(httpStatus.NOT_FOUND, 'Video not found');
+  }
+
+  if (updatedVideo.propertyId) {
+    const property = await getPropertyById(updatedVideo.propertyId).catch((error) => {
+      throw new ErrorHandler(httpStatus.INTERNAL_SERVER_ERROR, 'Internal Server Error', error);
+    });
+
+    if (!property) {
+      throw new ErrorHandler(httpStatus.NOT_FOUND, 'Invalid property');
+    }
+
+    if (updatedVideo.vendorId.toString() !== property.addedBy.toString()) {
+      throw new ErrorHandler(httpStatus.FORBIDDEN, 'You are not permitted to perform this action');
+    }
+  }
+
+  const videoPayload = {
+    ...video.toObject(),
+    ...updatedVideo,
+  };
+
+  try {
+    return Video.findByIdAndUpdate(video._id, videoPayload, { new: true });
+  } catch (error) {
+    throw new ErrorHandler(httpStatus.BAD_REQUEST, 'Error updating video', error);
+  }
+};
+
+export const approveVideo = async ({ videoId, adminId }) => {
+  const video = await getVideoById(videoId).catch((error) => {
+    throw new ErrorHandler(httpStatus.INTERNAL_SERVER_ERROR, 'Internal Server Error', error);
+  });
+  if (!video) {
+    throw new ErrorHandler(httpStatus.NOT_FOUND, 'Video not found');
+  }
+
+  try {
+    return Video.findByIdAndUpdate(
+      video._id,
+      { $set: { approved: true, approvedBy: adminId } },
+      { new: true },
+    );
+  } catch (error) {
+    throw new ErrorHandler(httpStatus.BAD_REQUEST, 'Error updating video', error);
+  }
+};
+
+export const deleteVideo = async (videoDetails) => {
+  const video = await getVideoById(videoDetails.videoId).catch((error) => {
+    throw new ErrorHandler(httpStatus.INTERNAL_SERVER_ERROR, 'Internal Server Error', error);
+  });
+  if (!video) {
+    throw new ErrorHandler(httpStatus.NOT_FOUND, 'Video not found');
+  }
+
+  const property = await getPropertyById(video.propertyId).catch((error) => {
+    throw new ErrorHandler(httpStatus.INTERNAL_SERVER_ERROR, 'Internal Server Error', error);
+  });
+
+  if (videoDetails.vendorId.toString() !== property.addedBy.toString()) {
+    throw new ErrorHandler(httpStatus.FORBIDDEN, 'You are not permitted to perform this action');
+  }
+
+  try {
+    return Video.findByIdAndDelete(video._id);
+  } catch (error) {
+    throw new ErrorHandler(httpStatus.BAD_REQUEST, 'Error deleting video', error);
+  }
+};
+
+export const getAllVideos = async (user, propertyId, { page = 1, limit = 10, ...query } = {}) => {
+  const property = await getPropertyById(propertyId).catch((error) => {
+    throw new ErrorHandler(httpStatus.INTERNAL_SERVER_ERROR, 'Internal Server Error', error);
+  });
+
+  if (!property) {
+    throw new ErrorHandler(httpStatus.NOT_FOUND, 'Invalid property');
+  }
+
+  if (user.role === USER_ROLE.VENDOR) {
```

Comment:
This can be merged into a single statement:

```suggestion
  if (user.role === USER_ROLE.VENDOR && user._id.toString() !== property.addedBy.toString()) {
```